std/typecons.d

```diff
@@ -6450,7 +6452,7 @@ if (!is(T == class) && !(is(T == interface)))
             {
                 pureGcRemoveRange(&this._store._payload);
             }
-            pureFree(_store);
+            () @trusted { pureFree(_store); }();
```

Comment:
I'd feel better if `deallocateStore` was in a trusted lambda where it is called rather than have an `@trusted` function that deallocates things

Reply:
I started off that way, then thought "when is this not memory-safe?". The reason for `free` itself to be `@system` is twofold: one can pass a pointer that wasn't allocated with malloc to it (such as to the stack), and the other is use-after-free problems if there are aliases to that memory elsewhere. 

I don't think either can happen here.

Reply:
Now that I think of it, it definitely can because there's no check here about the ref count.

Reply:
(unresolving conversation)

Reply:
I resolved it because I addressed the comment in the latest version.